webpack.config.js

```diff
@@ -1,18 +1,28 @@
 const path = require('path');
 const webpack = require('webpack');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
 
 module.exports = {
   entry: {
     app: './src-js/app.js'
   },
   output: {
-    path: path.join(__dirname, 'jekyll/assets/js/'),
-    filename: '[name].min.js'
+    // We need the flexibility to build a js file to one place and an html file
+    // in another, which are in different places in the jekyll directory tree.
+    // We set a fairly high level path and empty default path so that we can get
+    // this flexibility.
+    path: path.join(__dirname, 'jekyll/'),
+    publicPath: '',
+    filename: 'assets/js/dist/[name]-[contenthash].min.js'
   },
   plugins: [
-    new webpack.ProvidePlugin({
-      React: 'react',
-      ReactDOM: 'react-dom'
+    new HtmlWebpackPlugin({
+      filename: '_includes/app.html',
+      templateContent: ({ htmlWebpackPlugin }) =>
+        `<script src="{{ "/${
+          htmlWebpackPlugin.files.chunks.app.entry
+        }" | prepend: site.baseurl }}" type="text/javascript"></script>`,
```

Comment:
not sure what's happening here or why it's necessary. 

Looks like we're iterating over plugin files to include as JS. Is this the standard way to do things? Why have this template at all? We should just be including the generated webpack JS once in the default template, and including all other js files into the webpack... unless i'm missing something of course.